client/dashboard/sites/add-new-site/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { __, sprintf } from '@wordpress/i18n';
+import { download, reusableBlock, Icon } from '@wordpress/icons';
+import clsx from 'clsx';
+import devSiteBanner from 'calypso/assets/images/a8c-for-agencies/dev-site-banner.svg';
+import { preventWidows } from 'calypso/lib/formatting';
+import Column from './column';
+import MenuItem from './menu-item';
+import './style.scss';
+
+const TRACK_SOURCE_NAME = 'sites-dashboard';
+
+const wordpressClick = () => {
+	recordTracksEvent( 'calypso_sites_dashboard_new_site_action_click_add' );
+	recordTracksEvent( 'calypso_sites_dashboard_new_site_action_click_item', {
+		action: 'wordpress',
+	} );
+};
+const jetpackClick = () => {
+	recordTracksEvent( 'calypso_sites_dashboard_new_site_action_click_jetpack' );
+	recordTracksEvent( 'calypso_sites_dashboard_new_site_action_click_item', {
+		action: 'jetpack',
+	} );
+};
+const migrateClick = () => {
+	recordTracksEvent( 'calypso_sites_dashboard_new_site_action_click_item', {
+		action: 'migrate',
+	} );
+};
+const importClick = () => {
+	recordTracksEvent( 'calypso_sites_dashboard_new_site_action_click_import' );
+	recordTracksEvent( 'calypso_sites_dashboard_new_site_action_click_item', {
+		action: 'import',
+	} );
+};
+const offerClick = () => {
+	recordTracksEvent( 'calypso_sites_dashboard_new_site_action_click_item', {
+		action: 'offer',
+	} );
+};
+
+function AddNewSite() {
+	return (
+		<HStack alignment="flex-start" style={ { padding: '16px' } } spacing={ 4 }>
+			<Column title={ __( 'Add new site' ) }>
+				<MenuItem
+					icon={ <WordPressLogo /> }
+					title={ __( 'WordPress.com' ) }
+					description={ preventWidows(
+						__( 'Build and grow your site, all in one powerful platform.' )
+					) }
+					onClick={ wordpressClick }
+					href="/start?source=sites-dashboard&ref=new-site-popover"
+				/>
+				<MenuItem
+					icon={ <JetpackLogo /> }
+					title={ __( 'Via the Jetpack plugin' ) }
+					description={ preventWidows( __( 'Install the Jetpack plugin on an existing site.' ) ) }
+					onClick={ jetpackClick }
+					href={ `/jetpack/connect?cta_from=${ TRACK_SOURCE_NAME }&cta_id=add-site` }
+				/>
+			</Column>
+			<Column title={ __( 'Migrate and import' ) }>
+				<MenuItem
+					icon={ reusableBlock }
+					title={ __( 'Migrate' ) }
+					description={ preventWidows(
+						__( 'Bring your entire WordPress site to WordPress.com.' )
+					) }
+					onClick={ migrateClick }
+					href="/setup/site-migration?source=sites-dashboard&ref=new-site-popover&action=migrate"
+				/>
+				<MenuItem
+					icon={ <Icon icon={ download } size={ 18 } /> }
+					title={ __( 'Import' ) }
+					description={ preventWidows(
+						__( 'Use a backup to only import content from other platforms.' )
+					) }
+					onClick={ importClick }
+					href="/setup/site-migration/create-site?source=sites-dashboard&ref=new-site-popover&action=import"
+				/>
+			</Column>
+
+			<Button
+				// Figure out localizeUrl
+				href="https://wordpress.com/setup/onboarding"
```

Comment:
localizeUrl is defined in `i18n-utils` package. I don't really like "utils" packages but maybe that one would be ok to use. What do you think @jsnajdr ? What would be your recommendation.

Reply:
`localizeUrl` is a self-contained function with (almost) no dependencies, so it's fine to use it.

However, there are dependencies: it has the `locale` second parameter which defaults to `getDefaultLocale()`, and that introduces a dependency on `@wordpress/i18n` and its `i18n` global. That's not bad, we use `@wordpress/i18n` anyway.

I'd recommend to use the reactive `useLocalizeUrl` variant which reads `i18n` from context.

One unfortunate thing is the `localeContext` context provider, which you have to mount. The bad thing about it is that it has a huge overlap with the context in `@wordpress/react-i18n`. These two context should be merged together. At this moment you have to mount both.

Reply:
Thanks, just to recap, we discussed a bit in DM and it seems the add new site component is only used in the sites dashboard (v1 and v2) and both require the user to be logged in, this means that there's no need to localize these links at all.

Reply:
Yes, I got lost a bit in the (very interesting) details of `localizeUrl` and forgot to mention that you probably don't need it 😆 For a logged in user the primary source of the locale is the user account info. Specifying locale in the URL itself is mainly for logged-out users who are not yet signed up.